kqueen/auth/ldap.py

```diff
@@ -61,30 +92,42 @@ def verify(self, user, password):
             None: Invalid credentials, Authentication failure.
             User: authenticated user.
         """
+        matched_dn = self._get_matched_dn(user.username)
+
+        for dn in matched_dn:
+            if self._bind(dn, password):
+                full_dn = dn
+
+        if full_dn:
```

Comment:
You are defining `full_dn` variable inside `if` condition so it may not be defined if none of the matched users authenticate with given password, it would raise `referenced before assignment` error. You should prolly add `full_dn = None` above the `for` loop. Also I don't see you actually storing the verified `full_dn` on User object and checking for it in the beginning of `verify` method to skip the search.

Reply:
good point, thanks! about metadata storing, before my changes, i wanted to get your review and discuss, so it does not exist yet